File: backend/src/main/java/com/app/backend/security/PermissionCodes.java
package com.app.backend.security;

/**
 * Central permission codes (stored in DB and enforced server-side).
 */
public final class PermissionCodes {

    private PermissionCodes() {
    }

    public static final String OWNER_VIEW = "OWNER_VIEW";
    public static final String OWNER_EDIT = "OWNER_EDIT";
    public static final String PLOT_DELETE = "PLOT_DELETE";
    public static final String USER_MANAGE = "USER_MANAGE";
    public static final String ASSIGN_AGENT = "ASSIGN_AGENT";
    public static final String VIEW_ACTIVITY_LOGS = "VIEW_ACTIVITY_LOGS";
}

File: frontend/src/app/core/constants/permission-codes.ts
/** Mirrors backend {@code PermissionCodes} — UI hints only; authorization is enforced server-side. */
export const PERMISSION_CODES = {
  OWNER_VIEW: 'OWNER_VIEW',
  OWNER_EDIT: 'OWNER_EDIT',
  PLOT_DELETE: 'PLOT_DELETE',
  USER_MANAGE: 'USER_MANAGE',
  ASSIGN_AGENT: 'ASSIGN_AGENT',
  VIEW_ACTIVITY_LOGS: 'VIEW_ACTIVITY_LOGS',
} as const;

export type PermissionCode = (typeof PERMISSION_CODES)[keyof typeof PERMISSION_CODES];
